### DISCUSSION TITLE
SiliconDust Flex 4K

### DISCUSSION THREAD

**Discussion by techiegirl2day** [OpenSageTV/Community#32] [Tuners]
Adding new thread.  Anyone using this newer tuner? After my Connect Quad crapped out I went with this one hoping it is a bit future proof since it has both ATSC 1.0 & 3.0 tuners. As far as I know the 3.0 tuner specs are not in use in Canada, unless you are on US border & can access channels with both ATSC 1 or 3 specs.

**Comment by sic0048** [comment]
I'm currently using one with my SageTV set up.  It is receiving both ATSC 2.0 and ATSC 3.0 signals and I can record both in Sage.  That being said, none of the stations in my area are actually broadcasting in 4k right now on ATSC 3.0.  I'm not sure if Sage will support 4k recordings or not.

Stations aren't expected to broadcast in 4k until ATSC 2.0 is sunset.  Until then the stations don't have enough bandwidth to broadcast both ATSC 3.0 in 4k and ATSC 2.0 at the same time.  At least that is what they are saying.....

**Reply by wayner9** [reply]
In theory I think SageTV should be able to support 4K playback on devices that have 4K.  I think people have watched 4K content from downloaded movie/TV files in SageTV.

**Reply by techiegirl2day** [reply]
I know I have been experiencing picture playback freeze up on ATSC 1.0 which seems to be an issue on this device from what others have said in SiliconDust forum. I guess the data is too much for bandwidth to handle.

**Reply by techiegirl2day** [reply]
I am realizing it could be a router issue as well. My router is getting long on the tooth. I will have to wait till I can afford to upgrade to a newer one with bigger bandwidth/speed so I can test if it is a factor.

**Reply by wayner9** [reply]
You can get switches pretty cheaply and you may just need a switch since the path from your HDHR device to your SageTV PC will just be on your LAN and doesn't have to hit the router.  If you have a smaller network then your router is a router+switch+ wireless access point.  In more sophisticated networks those are separate devices.  I have separate devices for a router, I have about 5 or so switches on my network and six wireless access points.

Five port gigabit switches are as cheap as $15 and 8 port switches $20.

**Comment by GitOutWest** [comment]
I thought about adding an SiDust ATSC3.0 tuner (2x ATSC3.0, 2x ATSC1.0) to my mix; still running older SiDust 1K tuners with SageTV. Even though few stations in my (SFO Bay) area are broadcasting that way yet, and 4K TV isn't reported to look much different than 1K given the signal sources. We did have a shuffle a couple months ago as a stepping stone to some ATSC3.0, and had to rescan for the existing stations.

However, I'm now hearing about stations across the US where ATSC3.0 had been running now enabling encryption. I'm thinking that 1K free TV may be the end-of-the-line, with no free-open-TV system (HW & SW) supporting encrypted ATSC3.0 when it's forced on us.

https://cordcuttersnews.com/comcast-is-starting-to-encrypt-its-free-over-the-atsc-3-0-tv-stations-meaning-many-cord-cutters-cant-watch-free-local-tv-anymore-without-upgrade/

https://www.howtogeek.com/895901/nextgen-tv-and-atsc-3.0-is-not-going-well/

**Reply by eelpout** [reply]
whereabouts in the Bay Area, are you? I'm down near Los Gatos and I can't receive any NG stations from SF's Sutro Tower with my assortment of indoor antennas.

There is supposed to be a low-power ATSC 3.0 station on Loma Prieta peak, but I believe they are primarily using a special subcarrier developed for FM radio delivery (87.7). Could be I simply can't get the TV signal on 6.1/6.2 on my Connect 4K because my antennas are all oriented the other way. Or it's not actually transmitting.

I'd been to that tower site a few times back in the day when the NBC affiliate was still transmitting from there with an acquaintance who worked on the transmitter, that was before it was relegated to backup use.

**Comment by wayner9** [comment]
ATSC 1.0 is actually 1080x1920 or 1080i or 720p which is 2K vs "4K" which is actually 2160x3840 or 2160p.

Some original content is acutally in 4K and looks much better, especially sports.  Big events like the Superbowl are definitely done in 4K.

The transition to ATSC3.0 is going to take a while due to the bandwidth issue.  The FCC is not allocating new bandwidth for ATSC 3.0 and there isn't enough bandwidth to transmit both ATSC1 and 3 on the same feed.

**Comment by RJ-Make** [comment]
SageTV can definitively **_Record_** 4K (Ceton Tuner). I've done it many times when FiOS has their 4K events.  My problem is, Comskip can't handle 4K so my automated system fails. 

And yes, the ATSC3.0 DRM is more then likely going to kill the entire ecosystem.

**Reply by PhilipCRoberts** [reply]
CATV channels that die when you're watching (recording) them for more than 3 hours is bad enough.

**Reply by jwittkoski** [reply]
@RJ-Make Comskip supports 4K processing. See this commit: https://github.com/erikkaashoek/Comskip/commit/755249b8ab615909919d9a107dfb479a48623a72

I recompiled it (some time in 2022) for my Linux server and works on 4k channels like the FiOS 4K event channels. It's running on the 4K Colorado TCU game right now. 😄

**Reply by jwittkoski** [reply]
I will add that comskip for the 2h 48m 4K recording took about 18h to complete on my SageTV server. Midway through it had a bunch of commercial times in the txt file but when it finished there were none. So it runs (and doesn't crash) but isn't very useful. Perhaps I need different settings for 4K comskip.

**Comment by CptKirk-1701** [comment]
I live in Houston and have had Flex 4K for quite a while now.  There are plenty of channels being broadcast in Houston in ATSC 3.0.

ATSC 3.0 uses the H.265 HEVC codec for video and Dolby AC-4 for audio.

I can tune into the channels on my computer using the HDHomeRun Setup application.  When I click on a channel in their app it launches the video in VLC.

All of the Houston ATSC 3.0 channels show the following for the video:

Codec: MPEG-H Part2/HEVC (H.265) (hevc)
Video resolution: 1920x1080
Frame rate: 59.940060

BUT, there is NO audio.

There are no AC-4 audio drivers for Windows computers.  As far as I know (and I have searched a few times), there is no one working at creating AC-4 audio driver for Windows.

And yeah, there are already stories about stations putting DRM on even the standard ABC, CBS, NBC, etc ATSC 3.0 channels.

So for me the Flex 4K is just a 4 tuner ATSC 1.0 device.

**Comment by RJ-Make** [comment]
> @RJ-Make Comskip supports 4K processing. See this commit: [erikkaashoek/Comskip@755249b](https://github.com/erikkaashoek/Comskip/commit/755249b8ab615909919d9a107dfb479a48623a72)
> 
> I recompiled it (some time in 2022) for my Linux server and works on 4k channels like the FiOS 4K event channels. It's running on the 4K Colorado TCU game right now. 😄

Doesn't work for me on Windows. Every attempt just ends in Comskip crashing.

**Reply by jwittkoski** [reply]
Sorry to hear that. Which comskip version are you using? I saw on this post (http://www.kaashoek.com/comskip/viewtopic.php?f=5&t=1962) that comskip may have problems with ATSC 3.0, even at lower-than-4K resolutions. The OP has an example using ffmpeg and comskip that may allow comskip to run. Might be worth manually trying that with your recordings to see if there is a difference. I don't have ATSC 3.0 so can't verify with my setup.

**Reply by RJ-Make** [reply]
build .012

**Comment by GitOutWest** [comment]
Still using SageTV (now 9.2.10, on a Win11 PC) in 2025 with a 30-yr-old rooftop antenna. The family still really likes it compared to alternatives, on multiple TVs[PCs via HDMI] in the house.

I recently had to deal with a Schedules Direct EPG issue. And some flaky HW (several old Silicon Dust dual-tuners & power bricks; previously had to jump thru some OpenDCT hoops) that lead me to upgrade/replace/simplify those with a couple SiDust HDHomeRun Flex 4K tuners (2x ATSC-3.0, 2x ATSC-1.0; single antenna cable & power brick; works without OpenDCT). All is working fine.

I can see via HDHomeRun's software and the ATSC-3.0-capable tuners a few ATSC-3.0 signals in my (SFO Bay Area) area. They're useless since they're clustered as one "test freq" from one broadcaster, with that broadcaster simply rebroadcasting "low-rez" (2K, not 4K) signals from other area broadcasters as sub-channels within the 3.0-envelope. Of those, four of the six signals are DRM-encrypted--something that doesn't bode well for the future of OTA ATSC-3.0. But I *can* use the HDHomeRun software [on PC and on Android] + GUI-live + VLC (etc.) to view the unencrypted ATSC-3.0-encoded channels, although no-audio is a problem with latest VLC (OK with HDHR's software).

I haven't spent much time with Sage to see if I can get *it* to scan for & use (video *and* audio), and ideally record, any of the unencrypted ATSC-3.0 signals. 

Has anybody spent some time and tried this in 2025+?  
    e.g. SageTV 9.2.10 (etc.), Windows, SiDust HDHomeRun Flex 4K tuners, ATSC 3.0? 
Will Sage scan for and properly use those 3.0 signals and channels from the HDHR tuners? [or will I have to watch them some other way?]

**Comment by CptKirk-1701** [comment]
Please see my comment on Jun 17, 2023.  Everything I stated back then still applies for me in now Houston.

**Reply by GitOutWest** [reply]
I found articles saying AC-3 audio was in then out of Windows 11.

A couple of suggestions for bringing it back which I'll look into:

https://community.adobe.com/t5/premiere-pro-discussions/no-ac3-codec-in-windows-11-24h2-solution-is-needed/m-p/14972691
https://community.adobe.com/t5/premiere-pro-discussions/no-ac3-codec-in-windows-11-24h2-solution-is-needed/m-p/14974425#M537882
https://www.majorgeeks.com/files/details/dolby_ac_3ac_4_installer.html

https://github.com/Victor-Freeze/AC-3-for-Windows-11

**Comment by CptKirk-1701** [comment]
ATSC 3.0 uses Dolby AC-4 for audio, not AC-3.

**Reply by GitOutWest** [reply]
I haven't tried it yet, but that majorgeeks link says "Dolby AC-3/AC-4 Installer", so we'll see
https://www.majorgeeks.com/files/details/dolby_ac_3ac_4_installer.html

**Comment by cncb-gh** [comment]
I guess Dolby AC-4 audio decoding was never added?  What about HEVC video support?

**Reply by wayner9** [reply]
I think it depends on the playback.  These will not work on extenders, but I believe that you should be able to load these codecs for playback on PCs. I think that people have used HEVC for ripped movies.

**Comment by CptKirk-1701** [comment]
There is NOTHING out there in the ecosystem for decoding a Dolby AC-4 audio stream.  Due to licensing (ie Dolby wants money for everything), NOBODY has written a Windows Dolby AC-4 codec.

So you can tune into a ATSC 3.0 stream with no problems, just no going to get any audio.

I really don't trust anything MajorGeeks posts for you to download.  You have absolutely no idea what you will be installing on your system.

Maybe try it in a VM that you can wipe afterwards but not on a production machine.

The ffmpeg folks had a ticket open for a long time talking about creating the AC-4 codec.  But then they dropped it like a hot potato, and it never started up again.

**Reply by wayner9** [reply]
I did not know that.  I thought that there would be Windows decoders just like any other compression format.

Can you play back the audio on any devices?  What about TVs?  STBs like Roku, AppleTV or GoogleTV?

**Reply by SESNut** [reply]
there is one windows decoder, its literally from dolby and its meant for oems from the windows store like the hevc codec that no one wants to pay a dollar for because licensing. thats what the majorgeeks download is, its the app package from the windows store. You dont have to get it from there, you can just use a script to download it from the windows store

**Comment by cncb-gh** [comment]
I have been following ffmpeg and there are "unofficial" builds that decode AC-4 so if it is just a question of plugging in a new ffmpeg into Sage it might be worth a try.  The newest Fire Cube decodes AC-4 and the HDHomerun software passes it on untouched to the Cube for decoding.  Emby and Channels seem to be using software decoding.

**Reply by emveepee** [reply]
The main issue with ATSC 3.0 isn't really AC-4.  I've written client for NextPVR and some Roku and Android players support proper AC-4 decoding and the ffmpeg patch/hack at leaves partial support.  Tizen supports AC-4 in mp4.    The issue continues to be DRM and spending too much effort on a client seems to me largely a waste of time.

**Comment by cncb-gh** [comment]
DRM is the big problem, yes, but not the most pressing one for me at the moment.  The combining of channels for atsc3 seems to have caused me to not be able to tune my atsc1 abc channel.  So, I have to use atsc3 for abc which happens to still be drm-free. AC-4 decoding is the short-term problem for me. HDHomerun software is acceptable so I guess I will just keep using that for the time being.
